src/helpers.sh

```diff
@@ -239,7 +239,7 @@ function helpers::find_total_tests() {
                 for fn_name in "${functions_to_run[@]}"; do
                     local provider_data=()
```

Comment:
In Bash 3.0, this doesn't work. This is equivalent to `local provider_data='()'`.

Reply:
Thanks for the feedback. I didn't tested with real bash 3.0 binary as I had to leave after I created the PR and wanted to know how far away from the final result this would be. I will continue with it and test it with bash 3.0 binary this weekend. Also, I should find a way to setup this bash 3.0 in the CI to run all tests using that version